### PR TITLE
refactor handling input to support project files

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: src/cloud.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ function getMutationVariables(
   // GITHUB_REF_NAME is the correct branch when running in a branch, on pull requests it's the PR number.
   const sourceBranch =
     process.env.GITHUB_HEAD_REF || process.env.GITHUB_REF_NAME
-  const migrationDir = getMigrationDir(opts.dir).replace('file://', '')
+  const migrationDir = res.summary?.Env.Dir.replace('file://', '')
```
Comment (giautm) [inline src/cloud.ts:58]: This is not safe, it the `Env` /or `Dir` is undefined

File: __tests__/cloud.test.ts [hunk 2]
```diff
@@ -55,7 +55,10 @@ describe('report to cloud', () => {
       raw: '[{"Name":"test","Text":"test"}]',
       summary: {
         Files: [{ Name: 'test', Text: 'test' }],
-        Env: {},
+        Env: {
+          Driver: 'MySQL',
+          Dir: 'migrations/'
```
Comment (yonidavidson) [inline __tests__/cloud.test.ts:60]:
```suggestion
          Dir: 'migrations'
```

File: __tests__/cloud.test.ts [hunk 3]
```diff
@@ -85,9 +88,7 @@ describe('report to cloud', () => {
           commit: process.env.GITHUB_SHA,
           envName: 'CI',
           payload: '[{"Name":"test","Text":"test"}]',
-          projectName: `${process.env.GITHUB_REPOSITORY}/${getMigrationDir(
-            opts.dir
-          )}`,
+          projectName: `${process.env.GITHUB_REPOSITORY}/${res.summary?.Env.Dir}`,
```
Comment (yonidavidson) [inline __tests__/cloud.test.ts:91]:
```suggestion
          projectName: path.join(`${process.env.GITHUB_REPOSITORY},${res.summary?.Env.Dir}),
```

File: __tests__/cloud.test.ts [hunk 4]
```diff
@@ -146,7 +150,10 @@ describe('report to cloud', () => {
       raw: '[{"Name":"test","Text":"test"}]',
       summary: {
         Files: [{ Name: 'test', Text: 'test' }],
-        Env: {},
+        Env: {
+          Driver: 'MySQL',
+          Dir: 'migrations/'
```
Comment (yonidavidson) [inline __tests__/cloud.test.ts:155]:
```suggestion
          Dir: 'migrations'
```

File: __tests__/cloud.test.ts [hunk 5]
```diff
@@ -174,7 +181,10 @@ describe('report to cloud', () => {
...
```
Comment (yonidavidson) [inline __tests__/cloud.test.ts:186]:
```suggestion
          Dir: 'migrations'
```

File: __tests__/input.test.ts [hunk 6]
```diff
@@ -1,21 +1,64 @@
...
-import { createTestEnv, defaultVersion } from './env'
+import { createTestEnv, defaultEnv, defaultVersion } from './env'
+import { atlasArgs } from '../src/atlas'
 
 describe('input', () => {
-  test('defaults', async () => {
-    let prep = await createTestEnv()
-    let options = OptionsFromEnv(prep.env)
+  test('defaults', () => {
+    let options = OptionsFromEnv(defaultEnv)
```
Comment (yonidavidson) [inline __tests__/input.test.ts:8]:
```suggestion
    const options = OptionsFromEnv(defaultEnv)
```

File: __tests__/input.test.ts [hunk 7]
```diff
@@ -1,21 +1,64 @@
...
+      latest: 3,
+      projectEnv: 'env'
     }
     expect(options).toEqual(expected)
   })
 })
+
+describe('atlas args', () => {
+  test('without env', async () => {
+    let opts = OptionsFromEnv({
```
Comment (yonidavidson) [inline __tests__/input.test.ts:44]:
```suggestion
    const opts = OptionsFromEnv({
```

File: action.yml [hunk 8]
```diff
@@ -33,6 +28,9 @@ inputs:
     description: 'Enable reporting of schema insights (requires ariga-token)'
     default: 'true'
     required: false
+  project-env:
+    description: "The env from the atlas.hcl project file to use"
```
Comment (yonidavidson) [inline action.yml:32]: I think `The environment name` is enough, no need to couple it with where it came from

**Review by yonidavidson — APPROVED** [review]
Looks good - check out the few comments

**Pull request merged**